src/endpoints/mod_feedback.rs

```diff
@@ -0,0 +1,115 @@
+use actix_web::{get, post, web, HttpResponse, Responder};
+use serde::{Deserialize};
+use sqlx::Acquire;
+
+use crate::{
+    extractors::auth::Auth,
+    types::{
+        api::{ApiError, ApiResponse},
+        models::{
+            developer::{Developer},
+        },
+    },
+    AppData
+};
+use crate::types::models::mod_version::ModVersion;
+use crate::types::models::mod_feedback::{ModFeedback,FeedbackTypeEnum};
+
+#[derive(Deserialize)]
+pub struct GetModFeedbackPath {
+    id: String,
+    version: String
+}
+
+#[derive(Deserialize)]
+pub struct PostModFeedbackPayload {
+    feedback_type: FeedbackTypeEnum,
+    feedback: String,
+}
+
+#[get("/v1/mods/{id}/versions/{version}/feedback")]
+pub async fn get_mod_feedback(
+    data: web::Data<AppData>,
+    path: web::Path<GetModFeedbackPath>,
+    auth: Auth,
+) -> Result<impl Responder, ApiError> {
+    let dev = auth.developer()?;
+    let mut pool = data.db.acquire().await.or(Err(ApiError::DbAcquireError))?;
+
+    if !Developer::has_access_to_mod(dev.id, &path.id, &mut pool).await? && !dev.admin && !dev.verified {
```

Comment:
you should probably only call this once, and reuse the result, since every call is a db hit.

Reply:
Good idea.